File: src/types/steamFeatured.ts
export type SteamFeaturedItem = {
  app_id: number;
  name: string;

  header_image?: string | null;
  large_capsule_image?: string | null;
  small_capsule_image?: string | null;

  discounted: boolean;
  discount_percent?: number | null;
  original_price?: number | null;
  final_price?: number | null;
  currency?: string | null;

  platforms: string[];
};

export type SteamFeaturedCategory = {
  id: string;
  name: string;
  items: SteamFeaturedItem[];
};

File: src-tauri/src/models/steam_featured.rs
use serde::Serialize;

#[derive(Serialize)]
pub struct SteamFeaturedCategory {
    pub id: String,
    pub name: String,
    pub items: Vec<SteamFeaturedItem>,
}

#[derive(Serialize)]
pub struct SteamFeaturedItem {
    pub app_id: u32,
    pub name: String,

    pub header_image: Option<String>,
    pub large_capsule_image: Option<String>,
    pub small_capsule_image: Option<String>,

    pub discounted: bool,
    pub discount_percent: Option<i64>,
    pub original_price: Option<i64>,
    pub final_price: Option<i64>,
    pub currency: Option<String>,

    pub platforms: Vec<String>,
}